### PR TITLE
[Experimental] Add e2e tests for the rating filter block

### LINKED ISSUE #42253: E2E tests: Rating Filter
Convert tests to Playwright edition. Remove legacy test after conversion. Fix any flaky tests.

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/42253

Adds basic e2e tests for the rating filter block. These are basic smoke tests, but could be extended in future to cover more complex cases. 

Also a minor rename of a variable in the rating filter block class to match it's intent.

### How to test the changes in this Pull Request:

1. See the new e2e tests pass in CI

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

[Experimental] Add e2e tests for the new product rating filter block.

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/rating-filter.block_theme.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+test.describe( 'Product Filter: Rating Filter Block', async () => {
+	test.describe( 'frontend', () => {
+		test( 'Renders a checkbox list with the available ratings', async ( {
+			page,
+		} ) => {
+			await page.goto( '/product-filters-rating-block/' );
+			const ratingStars = page.locator(
+				'.wc-block-components-product-rating__stars'
+			);
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/rating-filter.block_theme.spec.ts]: Can we replace css class locator here (and below) with a role base or label base locator? This particular element has `aria-label` set to `Rated %d out of 5`. By doing so, if we refactor the markup of stars we don't need to update this test.

**Review by dinhtungdu — COMMENTED** [review]
The code looks good and the test is also passing. I just left a comment about the locator. Otherwise, this is LGTM.

**Comment by samueljseay** [comment]
@dinhtungdu I have refactored selectors to use labels and added opt-in aria label support to CheckboxList so it can also be selected. Can you take another look?

**Review by dinhtungdu — DISMISSED** [review]
@samueljseay thank you for the update 🙇🏽 ! This is looking good to me. :shipit:

**Comment by dinhtungdu** [comment]
Looks like the new rating added in this PR affects other tests.

image

**Comment by samueljseay** [comment]
@dinhtungdu thanks for spotting that. Will fix.

**Comment by samueljseay** [comment]
@dinhtungdu could you give this another review? I ended up having to change the fixtures and assertions a little bit. i did my best to make them better and less dependent on things like exact array index in the data thats there, but i think in future the real solution would be having these fixtures defined in a json or js file and actually having that file used to create the fixtures.

**Review by dinhtungdu — APPROVED** [review]
> but i think in future the real solution would be having these fixtures defined in a json or js file and actually having that file used to create the fixtures.

I agree. Thanks for your extra thoughts. Can we also consider using REST API to create reviews at test run time (and remove them after)?

With the plan and the test is passing, this PR looks good to me!

**Comment by samueljseay** [comment]
> I agree. Thanks for your extra thoughts. Can we also consider using REST API to create reviews at test run time (and remove them after)?

Yes I think now after assessing how easy it is to use rest API we should refactor anything using fixtures to use the rest api to set/tear them down. I think we can do that in future PRs.

**Pull request merged**